src/trans.c

```diff
@@ -1500,16 +1500,23 @@ Obj FuncON_KERNEL_ANTI_ACTION (Obj self, Obj ker, Obj f, Obj n) {
   UInt    deg, i, j, rank, len;
   Obj     out;
 
-  if (INT_INTOBJ(ELM_LIST(ker, LEN_LIST(ker))) == 0) {
+  assert(IS_LIST(ker) && IS_INTOBJ(n));
```

Comment:
I would prefer to split this in two separate assertions; that way one immediately knows which of the two was violated.